lib/json/stream/buffer.rb

```diff
@@ -29,7 +29,7 @@ def initialize
       def <<(data)
         # Avoid state machine for complete UTF-8.
         if @buffer.empty?
-          data.force_encoding(Encoding::UTF_8)
+          (+data).force_encoding(Encoding::UTF_8)
```

Comment:
Does this work?
See also: https://github.com/dgraham/json-stream/pull/8#discussion_r1897004752

Reply:
If `data` is frozen, `(+data).force_encoding(Encoding::UTF_8)` does nothing because the `(+data).force_encoding(Encoding::UTF_8)` result is ignored. The below `if data.valid_encoding?` check may be done with non UTF-8 encoding.

If `data`'s encoding is always UTF-8, we don't need this `(+data).force_encoding(Encoding::UTF_8)` entirely.

If `data`'s encoding may not be UTF-8, we need `data = (+data).force_encoding(Encoding::UTF_8)` or `data = +data; data.force_encoding(Encoding::UTF_8)`.

Reply:
@kou Ah, that is correct, thank you! I missed the required reassignment. Fixed in 831698f.